airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/errors/messages/MysqlErrorMessage.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.base.errors.messages;
+
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_DB_NAME_OR_USER_ACCESS_DENIED;
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_HOST_OR_PORT;
+import static io.airbyte.integrations.base.errors.utils.ConnectionErrorType.INCORRECT_USERNAME_OR_PASSWORD;
+import static io.airbyte.integrations.base.errors.utils.ConnectorName.MYSQL;
+
+import io.airbyte.integrations.base.errors.utils.ConnectorName;
+
+public class MysqlErrorMessage extends ErrorMessage {
+
+  {
+    CONSTANTS.put("28000", INCORRECT_USERNAME_OR_PASSWORD);
```

Comment:
I'm not sure these error codes can be generalized. Based on this reference from the [MySQL Error SQLSTATE](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-error-sqlstates.html) page it seems that `42000` can be either any number of the 123 different error states. Similarly `08S01` can any number of the 20 different error states, right?

Reply:
> I'm not sure these error codes can be generalized. Based on this reference from the [MySQL Error SQLSTATE](https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-error-sqlstates.html) page it seems that `42000` can be either any number of the 123 different error states. Similarly `08S01` can any number of the 20 different error states, right?

totally agree. We should not generalize error messages based on SqlState and use database error codes for all jdbc connectors

Reply:
How did these specific error codes get chosen originally? Also, I'm curious since there seems to be some specific overlap between error codes across the different database variants if some of these error codes can be placed into the `ErrorMessage` abstract class so it can be _shared_ across all databases

When the code currently catches a `SQLException` from a database, does that `Exception` not already have the relevant information needed for the user to debug?